codex-rs/app-server/README.md

```diff
@@ -32,14 +32,22 @@ Example:
 ## Core primitives
 
 We have 3 top level primitives:
-- Thread - a conversation between the Codex agent and a user. Each thread contains multiple turns.
-- Turn - one turn of the conversation, typically starting with a user message and finishing with an agent message. Each turn contains multiple items.
-- Item - represents user inputs and agent outputs as part of the turn, persisted and used as the context for future conversations.
+- **Thread** - a conversation between the Codex agent and a user. Each thread contains multiple turns.
+- **Turn** - one turn of the conversation, typically starting with a user message and finishing with an agent message. Each turn contains multiple items.
+- **Item** - represents user inputs and agent outputs as part of the turn, persisted and used as the context for future conversations.
 
 ## Thread & turn endpoints
 
 The JSON-RPC API exposes dedicated methods for managing Codex conversations. Threads store long-lived conversation metadata, and turns store the per-message exchange (input → Codex output, including streamed items). Use the thread APIs to create, list, or archive sessions, then drive the conversation with turn APIs and notifications.
 
+### Lifecycle overview
```

Comment:
Thanks, this is certainly an improvement, but we should consider a Mermaid diagram a la https://modelcontextprotocol.io/specification/2025-06-18/basic/lifecycle